compose/service.py

```diff
@@ -936,20 +937,55 @@ def build_volume_binding(volume_spec):
     return volume_spec.internal, "{}:{}:{}".format(*volume_spec)
 
 
+def ensure_engine_compatibility(external_path, internal_path):
```

Comment:
This was a tough one to name, I was nearly going to invent a word :crystal_ball: but I chatted with @aanand about it as well. I originally was going to use `transform_for_engine` or something similar so that it is very explicit that we are changing the way these paths look for the specific need of the engine.
`normalize_host_path` doesn't capture the specific need of we are doing this for the engine. I understand the point that there are larger issues than paths, could be `ensure_paths_are_engine_compatible` is clearer?